### ISSUE TITLE
Very slow evementwise scores evaluation

### ISSUE THREAD

**Issue by rth** [ramp-kits/solar_wind#17]
Computing the eventwise scores evaluation appears to be very time consuming, mostly due to the repeated calls to `pd.to_dataframe` within the `Eventwise*` scores. 

At least for `ramp_test_submission --quick-test` on the starting kit this accounts for most of the runtime,

![Screenshot_2019-03-29 out](https://user-images.githubusercontent.com/630936/55198750-960fe580-51b7-11e9-942e-8085b0860c4a.png)

Originally the starting kit with the `--quick-test` option takes 63s on my laptop, with evenwise scores disables this reduces to 9s.

Looking for a way speeding it, but generally such long run time (on a dataset that is not that big) makes iterations slower which is problematic when running events.

**Comment by glemaitre** [comment]
ping @jorisvandenbossche

**Comment by rth** [comment]
To give more details, it's running `pd.to_datetime(y_true[:, 0], unit='m')`, where `y_true[:, 0]` is a float in minutes. I think converting it to a timestamp (e.g. ns) and than back to datetype, without passing by `pd.to_datetime` might be faster, but I have not found a vectorized way to do that yet.

**Comment by jorisvandenbossche** [comment]
Yes, we recently had an issue about that on the pandas issue tracker (it might be fixed in the latest pandas release). 
In older releases, when you had floats, we are taking a very slow generic object path, while for ints it is optimized:

```
In [9]: a = np.arange(100000.)                

In [10]: %timeit pd.to_datetime(a, unit='m') 
593 ms ± 6.02 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [11]: %timeit pd.to_datetime(a.astype(int), unit='m')   
2.11 ms ± 44.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

I don't fully recall, but it might be we can simply cast to ints there? (I mean, the fact that it is float is maybe just because it was concatenated into a 2D array with actual floats, but those dates are originally ints?)

**Comment by jorisvandenbossche** [comment]
Yes, they are ints:

https://github.com/ramp-kits/solar_wind/blob/0156adca51406421701eb98262e2b6671e9ac345/problem.py#L87

So doing a `astype(int)` in the to_datetime call should be safe and speed up a lot.

**Comment by jorisvandenbossche** [comment]
See https://github.com/ramp-kits/solar_wind/pull/18

**Comment by rth** [comment]
Thanks @jorisvandenbossche ! Looks great!

**Comment by rth** [comment]
Let's keep this open for now, even if the to_datatime is a major improvement -- I'll try see if caching some of calculations in event-wise scores could improve performance more.
